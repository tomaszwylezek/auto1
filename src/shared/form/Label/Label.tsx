import React from 'react';

// Models
import { ILabelProps } from './Label.types';

export const Label: React.FC<ILabelProps> = ({ children, id, className }): JSX.Element => (
  <label htmlFor={id} className={className}>
    {children}
  </label>
);
