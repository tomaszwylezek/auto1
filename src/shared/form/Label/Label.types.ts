export interface ILabelOwnProps {
  id: string;
  className?: string;
}

export interface ILabelProps extends ILabelOwnProps {}
