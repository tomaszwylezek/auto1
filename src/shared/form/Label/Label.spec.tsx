import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';

import { Label } from './Label';

// Models

import { ILabelProps } from './Label.types';

describe('Label component should', (): void => {
  it('renders correctly', (): void => {
    const props: ILabelProps = {
      id: 'foo',
      className: 'foobar'
    };

    const component: ShallowWrapper<ILabelProps> = shallow(<Label {...props}>foo</Label>);

    expect(component).toMatchSnapshot();
  });
});
