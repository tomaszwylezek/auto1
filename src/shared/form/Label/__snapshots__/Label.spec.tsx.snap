// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label component should renders correctly 1`] = `
<label
  className="foobar"
  htmlFor="foo"
>
  foo
</label>
`;
